### PR TITLE
Add gc & index file generator

### PR DESCRIPTION
**Pull request by rustamwin**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | -

### CODE REVIEW

**Comment by rustamwin** [comment]
Partially replaces #47

File: src/Storage/FileStorage.php [hunk 1]
```diff
@@ -51,8 +60,87 @@ public function flush(): void
 
             $jsonObjects = $varDumper->asJsonObjectsMap();
             $this->filesystem->write($this->path . '/' . $this->idGenerator->getId() . '.obj.json', $jsonObjects);
+
+            $this->updateManifest();
```
Comment (samdark) [inline src/Storage/FileStorage.php:64]: I'd call it "index". It reflects the purpose better.

Comment (rustamwin) [reply src/Storage/FileStorage.php]: Alright.

File: src/Collector/IndexCollectorInterface.php [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+/**
+ * Index data collector responsibility is to collect data during application lifecycle.
+ */
+interface IndexCollectorInterface
+{
+    /**
+     * @return array data indexed
+     */
+    public function getIndexed(): array;
```
Comment (samdark) [inline src/Collector/IndexCollectorInterface.php:16]: That is significantly different from EventCollectorInterface. Is that intended?

Comment (samdark) [reply src/Collector/IndexCollectorInterface.php:16]: Could be renamed to `getIndexData` or `collectIndexData`.

Comment (rustamwin) [reply src/Collector/IndexCollectorInterface.php]: > That is significantly different from EventCollectorInterface. Is that intended?

Yes, there is only the number of events that can be indexed.

> Could be renamed to getIndexData or collectIndexData.

Renamed to `getIndexData`.

**Comment by samdark** [comment]
👍

**Pull request merged**